### DISCUSSION TITLE
Capacitator Series

### DISCUSSION THREAD

**Discussion by EarthShaping** [SpiceSharp/SpiceSharp#201] [General]
I am not getting why this circuit raise an error

var ckt = new Circuit(
               new VoltageSource("V1", "in", "0", pulse),
               new Resistor("R1", "in", "middle", 10),
               new Capacitor("CL", "middle", "out", 0.2),
               new Capacitor("CW", "out", "0", 0.4)
           );
 var tran = new Transient("tran", samplingTime_s, 10);
 tran.Run(ckt);

But if I put a resistance R2 in parallel to CL capacitator, it works

var ckt = new Circuit(
               new VoltageSource("V1", "in", "0", pulse),
               new Resistor("R1", "in", "middle", 10),
               new Resistor("R2", "middle", "out", double.MaxValue),
               new Capacitor("CL", "middle", "out", 0.2),
               new Capacitor("CW", "out", "0", 0.4)
           );
 var tran = new Transient("tran", samplingTime_s, 10);
 tran.Run(ckt);

What I am doing wrong?

**Comment by marcin-golebiowski** [comment]
Can you share the error?

Maybe its related to the issue #119

**Reply by EarthShaping** [reply]
It seems the same issue

The RuleViolation is:

![image](https://user-images.githubusercontent.com/5947051/166117747-da56143b-caac-486f-bfcc-5fbe3bbf7ea9.png)

**Reply by marcin-golebiowski** [reply]
Please try a code below.
If it doesn't help you, please let me know.

```csharp
            double pulse = 1.0;
            var ckt = new Circuit(
                new VoltageSource("V1", "in", "0", pulse),
                new Resistor("R1", "in", "middle", 1 * 1000), // 1k resistor
                new Capacitor("CL", "middle", "out", 0.001), // capacitor
                new Capacitor("CW", "out", "0", 0.001)  // capacitor
            );
            var tran = new Transient("tran", 0.01,  1, 0.01);  
            tran.TimeParameters.InitialConditions["middle"] = 0.0;
            tran.TimeParameters.InitialConditions["out"] = 0.0;
            tran.ExportSimulationData += (sender, args) =>
            {
                Console.WriteLine(args.GetVoltage("middle"));
                Console.WriteLine(args.GetVoltage("out"));
                Console.WriteLine("-");
            };
            tran.Run(ckt);
```

**Reply by EarthShaping** [reply]
It works. Thanks.
